### PR TITLE
refactor(api): define an enum for the XOR URL base encoding parameter

### LINKED ISSUE #258: `Safe::xorurl_base` should be an enum instead of a string
Was reading https://forum.safedev.org/t/understanding-the-safe-cli-rust-code/2612 when I noticed that the `Safe::xorurl_base` member is a string, even though only a few different values are valid for it. In cases where the range of values is restricted, an enum is a more appropriate choice. The input to `--xorurl` should be converted to this enum as soon as possible, instead of at the last moment as is done [here](https://github.com/maidsafe/safe-cli/blob/c373f976262ed5e1b45de88aff535f210f603742/src/api/xorurl.rs#L369).

Looks like there is already a `Base` enum which I would think is what should be stored in `Safe`.

### PR DESCRIPTION
**Pull request by bochaco**
Resolves #258

### CODE REVIEW

File: safe-api/src/api/keys.rs [hunk 1]
```diff
@@ -657,7 +657,7 @@ mod tests {
         use rand::distributions::Alphanumeric;
         use rand::{thread_rng, Rng};
         use unwrap::unwrap;
-        let mut safe = Safe::new("base32z");
+        let mut safe = Safe::new(None);
```
[the remark this answers is not in the document]
Comment (bochaco) [reply safe-api/src/api/keys.rs]: ah nice! I'll send a commit for that @m-cat

**Comment by bochaco** [comment]
Totally agree, I'll try to disable them altogether, in general I don't like these type of rules TBH

**Comment by m-cat** [comment]
> Totally agree, I'll try to disable them altogether, in general I don't like these type of rules TBH

I think this rule is actually pretty cool, since it forces unsafe functions to be documented with *why* they are unsafe, and instructions to the caller for being careful. But for FFI modules it is unnecessary to provide this documentation for every single function.

**Comment by bochaco** [comment]
yeah...but if you are not willing to document them for any reason, or like in our case doesn't help much doing it for each function, then it's just annoying it's by default enabled...dunno...

**Comment by m-cat** [comment]
Yeah, but the Rust compiler pestering you is also annoying! The tradeoff for the hassle here is that you end up with much more safe code where the lint *does* apply. That's worth it, in my opinion. But you're not alone in your opinion, there are even core members of the Rust team that do not use clippy.

**Pull request merged**